From 9bfdea880fe433f1faf6db95b0d31f64feb26c90 Mon Sep 17 00:00:00 2001
From: "jie.yuan" <jie.yuan@amlogic.com>
Date: Tue, 6 Apr 2021 18:41:52 +0800
Subject: [PATCH] vold: try more times when format private partition[1/1]

PD#SWPL-44712

Problem:
use storage menu format u-disk failed

Solution:
try more times when format private partition

Verify:
locally

Change-Id: Ifac99b7a94c46033aeae34d7d302fccd42f0a4ab
Signed-off-by: jie.yuan <jie.yuan@amlogic.com>
---
 model/PrivateVolume.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/model/PrivateVolume.cpp b/model/PrivateVolume.cpp
index 1875b7b..12d77fa 100644
--- a/model/PrivateVolume.cpp
+++ b/model/PrivateVolume.cpp
@@ -226,6 +226,15 @@ status_t PrivateVolume::doFormat(const std::string& fsType) {
     }
 
     if (resolvedFsType == "ext4") {
+        int retry = 8;
+        while (access(mDmDevPath.c_str(), F_OK)) {
+            retry--;
+            usleep(50000);
+            if (!retry) {
+                LOG(INFO) << "dm dev: " << mDmDevPath << " doesn't exist";
+                break;
+            }
+        }
         // TODO: change reported mountpoint once we have better selinux support
         if (ext4::Format(mDmDevPath, 0, "/data")) {
             PLOG(ERROR) << getId() << " failed to format";
-- 
2.26.1

